catalogue/webapp/pages/works.tsx

```diff
@@ -87,25 +91,140 @@ const Works: NextPage<Props> = ({
+      id: 'production.dates.to',
+      value: productionDatesTo ?? undefined,
+    },
+  };
+
+  const workTypeFilter: CheckboxFilter = {
+    type: 'checkbox',
+    key: 'workType',
+    id: 'workType',
+    label: 'Formats',
+    options:
+      works?.aggregations?.workType.buckets.map((bucket) => ({
+        id: bucket.data.id,
+        value: bucket.data.id,
+        count: bucket.count,
+        label: bucket.data.label,
+        selected: worksRouteProps.workType.includes(bucket.data.id),
+      })) || [],
+  };
+
+  const subjectsFilter: CheckboxFilter = {
+    type: 'checkbox',
+    key: 'subjectsLabel',
+    id: 'subjects.label',
+    label: 'Subjects',
+    options:
+      works?.aggregations?.subjects?.buckets.map((bucket) => ({
+        id: toHtmlId(bucket.data.label),
+        value: quoteVal(bucket.data.label),
+        count: bucket.count,
+        label: bucket.data.label,
+        selected: worksRouteProps.subjectsLabel.includes(bucket.data.label),
+      })) || [],
+  };
+
+  const genresFilter: CheckboxFilter = {
+    type: 'checkbox',
+    key: 'genresLabel',
+    id: 'genres.label',
+    label: 'Genres',
+    options:
+      works?.aggregations?.genres?.buckets.map((bucket) => ({
+        id: toHtmlId(bucket.data.label),
+        value: quoteVal(bucket.data.label),
+        count: bucket.count,
+        label: bucket.data.label,
+        selected: worksRouteProps.genresLabel.includes(bucket.data.label),
+      })) || [],
+  };
+
+  const contributorsFilter: CheckboxFilter = {
+    type: 'checkbox',
+    key: 'contributorsAgentLabel',
+    id: 'contributors.agent.label',
+    label: 'Contributors',
+    options:
+      works?.aggregations?.contributors?.buckets.map((bucket) => ({
+        id: toHtmlId(bucket.data.agent.label),
+        value: quoteVal(bucket.data.agent.label),
+        count: bucket.count,
+        label: bucket.data.agent.label,
+        selected: worksRouteProps.contributorsAgentLabel.includes(
+          bucket.data.agent.label
+        ),
+      })) || [],
+  };
+
+  const languagesFilter: CheckboxFilter = {
+    type: 'checkbox',
+    key: 'languages',
+    id: 'languages',
+    label: 'Languages',
+    options:
+      works?.aggregations?.languages?.buckets.map((bucket) => ({
+        id: bucket.data.id,
+        value: bucket.data.id,
+        count: bucket.count,
+        label: bucket.data.label,
+        selected: worksRouteProps.languages.includes(bucket.data.id),
+      })) || [],
+  };
+
+  const locationsFilter: CheckboxFilter = {
+    type: 'checkbox',
+    key: 'itemsLocationsType',
+    id: 'items.locations.type',
+    label: 'Locations',
+    options:
+      works?.aggregations?.locationType?.buckets.map((bucket) => ({
+        id: bucket.data.type,
+        value: bucket.data.type,
+        count: bucket.count,
+        label: bucket.data.label,
+        selected: worksRouteProps.itemsLocationsType.includes(bucket.data.type),
+      })) || [],
+  };
+
+  const filters = [
```

Comment:
Perhaps these could live in a separate file that provides `getFilters(works)` or something like that? `works.tsx` is a bit of a monster

Reply:
I personally don't have a problem with working in the same file (as in the filters will only ever change at this point and aren't shared anywhere), I often prefer it, but if you think it makes it more readable, I'll move the filters (types and all) into it's own thing.

Reply:
I think I'd probably come down on the side of having them in a separate file. But I wouldn't be too upset if they weren't.

Reply:
These have been moved into the [`services/catalogue/filters`](https://github.com/wellcomecollection/wellcomecollection.org/pull/6101/files#diff-2c7db8d4fc4625288df1d7d3db38c386271720e8e06b8efca385dd47dabcee41) file